source/shared/core_sqlsrv.h

```diff
@@ -1438,8 +1438,10 @@ struct sqlsrv_stmt : public sqlsrv_context {
     bool has_rows;                        // Has_rows is set if there are actual rows in the row set
     bool fetch_called;                    // Used by core_sqlsrv_get_field to return an informative error if fetch not yet called 
     int last_field_index;                 // last field retrieved by core_sqlsrv_get_field
-    bool past_next_result_end;            // core_sqlsrv_next_result sets this to true when the statement goes beyond the 
-                                          // last results
+    bool past_next_result_end;            // core_sqlsrv_next_result sets this to true when the statement goes beyond the last results
+    bool columns_rows_obtained;           // Whether or not SQLNumResultCols and SQLRowCount have been called for the active result
+    short column_count;                   // Number of columns in the current result set obtained from SQLNumResultCols
+    long row_count;                       // Number of rows in the current result set obtained from SQLRowCount
     unsigned long query_timeout;          // maximum allowed statement execution time
```

Comment:
What's the chance of column_count or row_count ever be negative? If not, consider setting them to -1 (or any negative constant) such that you don't need the boolean variable `columns_rows_obtained`. This makes calling the two ODBC APIs independent of each other

Reply:
row_count can be -1 because SQLRowCount returns -1 if it can't tell how many rows are in the result. column_count is never negative but is a SQLSMALLINT so not unsigned.

Reply:
In this case, consider to initialize them to any negative constant(s). My suggestions are ACTIVE_RESULT_NUM_COLS_INVALID (-99) and/or ACTIVE_RESULT_NUM_ROWS_INVALID (-99). Something like that.